src/wallet/wallet.cpp

```diff
@@ -2150,6 +2150,14 @@ bool CWallet::SetCryptedHDSeed(const uint256& seedFp, const std::vector<unsigned
     return false;
 }
 
+HDSeed CWallet::GetHDSeedForRPC() const {
+    HDSeed seed;
+    if (!pwalletMain->GetHDSeed(seed)) {
+        throw JSONRPCError(RPC_WALLET_ERROR, "HD seed not found");
```

Comment:
Hmm, it just occurred to me that this, along with two other recent changes, blurs the boundary between `CWallet` and the RPC methods that use it. In the interest of moving towards `CWallet` being self-contained (and potentially a future swappable component), I don't think we should be using JSON-RPC code in `src/wallet/wallet.cpp`. However, given that we have already merged two changes that do so, I won't block on this here, as long as we don't add any more such code and make an issue to refactor this.

Reply:
I agree. I created an issue for this.

Reply:
Also this will break builds without ENABLE_WALLET, won't it?

Reply:
That is a good question. I just tested this with `ENABLE_WALLET` set to false and everything built fine.

Reply:
The wallet RPCs are removed when `ENABLE_WALLET` is set to false, so it isn't affected by this change.